Source/WebCore/page/Quirks.cpp

```diff
@@ -1618,6 +1618,21 @@ bool Quirks::shouldUseEphemeralPartitionedStorageForDOMCookies(const URL& url) c
     return false;
 }
 
+#if PLATFORM(IOS_FAMILY)
+// m365.cloud.microsoft rdar://157794706
+// Allow popups from m365.cloud.microsoft to onedrive.live.com
+bool Quirks::shouldAllowPopupFromMicrosoftOfficeToOneDrive(const URL& targetURL) const
+{
+    if (!needsQuirks())
+        return false;
+
+    if (m_document->url().host().endsWithIgnoringASCIICase("m365.cloud.microsoft"_s) && targetURL.host().endsWithIgnoringASCIICase("onedrive.live.com"_s))
+        return true;
+
+    return false;
+}
+#endif
+
```

Comment:
You should also add this to avoid doing more work than needed for the typical use case:

```
bool Quirks::needsPopupFromMicrosoftOfficeToOneDrive(const URL& targetURL) const
{
    if (!needsQuirks())
        return false;

    return m_quirksData.shouldAllowPopupFromMicrosoftOfficeToOneDrive;
}
```